test/unit/org/apache/cassandra/index/sasi/SASIIndexTest.java

```diff
@@ -137,7 +146,80 @@ public void cleanUp()
     }
 
     @Test
-    public void testSingleExpressionQueries()
+    public void testSASIComponentsAddedToSnapshot() throws Throwable
+    {
+        String snapshotName = "sasi_test";
+        Map<String, Pair<String, Integer>> data = new HashMap<>();
+        Random r = new Random();
+
+        for (int i = 0; i < 100; i++)
+            data.put(UUID.randomUUID().toString(), Pair.create(UUID.randomUUID().toString(), r.nextInt()));
+
+        ColumnFamilyStore store = loadData(data, true);
+        store.forceMajorCompaction();
+
+        try
+        {
+            Set<SSTableReader> ssTableReaders = store.getLiveSSTables();
+            Set<Component> sasiComponents = new HashSet<>();
+
+            for (Index index : store.indexManager.listIndexes())
+                if (index instanceof SASIIndex)
+                    sasiComponents.add(((SASIIndex) index).getIndex().getComponent());
+
+            Assert.assertFalse(sasiComponents.isEmpty());
```

Comment:
Nit: this could be out of the try-finally block, so the first statement of the block is taking the snapshot